File: public/dart-pkg/fuchsia/lib/fuchsia.dart
// Copyright 2016 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

library fuchsia;

import 'dart:isolate';
import 'dart:zircon';

// ignore_for_file: native_function_body_in_non_sdk_code

Handle _environment;
Handle _outgoingServices;

class MxStartupInfo {
  static Handle takeEnvironment() {
    Handle handle = _environment;
    _environment = null;
    return handle;
  }

  static Handle takeOutgoingServices() {
    Handle handle = _outgoingServices;
    _outgoingServices = null;
    return handle;
  }
}

void _setReturnCode(int returnCode) native "SetReturnCode";

Error exit(int returnCode) {
  _setReturnCode(returnCode);
  Isolate.current.kill(priority: Isolate.IMMEDIATE);
}

File: src/storage/fxfs/src/platform/fuchsia.rs
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

pub mod component;
mod device;
mod directory;
mod errors;
pub mod file;
pub mod log;
mod memory_pressure;
pub mod node;
pub mod pager;
mod remote_crypt;
pub mod vmo_data_buffer;
mod volume;
mod volumes_directory;

#[cfg(test)]
mod testing;

pub use remote_crypt::RemoteCrypt;
